File: Sangeun/src/week1/week8_problem_1.py
# https://programmers.co.kr/learn/courses/30/lessons/42577

def solution(phone_book):
    phone_book.sort(key=lambda x:len(x), reverse=True)

    while len(phone_book) > 1:
        tmp_num = phone_book.pop()
        tmp_result = [cmp_num for cmp_num in phone_book if cmp_num.startswith(tmp_num)]
        if len(tmp_result) > 0:
            return False
    return True

File: Shinhang/src/week1/week8_problem_1.java
package week1;

import java.util.*;

class Solution {
    public boolean solution(String[] phone_book) {
        boolean answer = true;
        Arrays.sort(phone_book);
        if(phone_book.length == 1){
            return answer;
        }
        for(int i = 0; i < phone_book.length - 1; i++){
            for(int j = i + 1; j < phone_book.length; j++) {
                if(i != j){
                    if(phone_book[j].startsWith(phone_book[i])){
                        return false;
                    }
                }
            }
        }
        return answer;
    }
}